app.js

```diff
@@ -234,25 +235,26 @@ async function setFinalReferenceValue(poolId, geostatsData) {
 
 }
 
-function isValidRequest(requestData) {
+function isValidRequest(requestData, poolExpiryTime) {
 
     agg_x = requestData.agg_x
     dataset = requestData.dataset
     band = requestData.band
     scale = requestData.scale
-    start_date = requestData.start_date
-    end_date = requestData.end_date
+    start_date_string = requestData.start_date
+    end_date_string = requestData.end_date
     geometry = requestData.geometry
 
+    pool_expiry_date_string = (new Date(parseInt(poolExpiryTime) * 1000)).toISOString().split('T')[0];
 
     if (agg_x == "agg_mean" || agg_x == "agg_max" || agg_x == "agg_min" || agg_x == "agg_median" || agg_x == "agg_stdDev" || agg_x == "agg_variance") {
-        if (dataset == "MODIS/006/MOD13Q1" && band == "NDVI" && scale == "250") {
+        if (dataset == "MODIS/MOD09GA_006_NDVI" && band == "NDVI" && scale == "463.313") {
             today = new Date()
             tenDaysAgo = new Date(new Date().setDate(today.getDate() - 10))
-            start_date = new Date(start_date)
-            end_date = new Date(end_date)
+            start_date = new Date(start_date_string)
+            end_date = new Date(end_date_string)
 
-            if (end_date < tenDaysAgo && start_date < end_date) {
+            if (start_date < tenDaysAgo && start_date < end_date && end_date_string == pool_expiry_date_string) {
```

Comment:
Should `start_date < tenDaysAgo && start_date < end_date` be replaced with `end_date - start_date >= 10 days`

Reply:
As agreed, let's change the distance to 30 days: `end_date - start_date >= 30 days`

Reply:
`if ((end_date - start_date) >= 30 days && (pool_expiry_date_string - end_date_string) >= 10 days)`

Reply:
I am thinking whether we also need a condition for `start_date > createdAt` where `createdAt` can be queried from the DIVA subgraph and represents the pool creation time. So if create a product, i would put the start_date at leasts a few minutes in the future to pass this check